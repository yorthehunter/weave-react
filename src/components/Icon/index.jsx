import React from 'react';
import { oneOfType, number, string } from 'prop-types';
import { paths, points } from './paths';

export const Icon = (props) => {
  const {
    width,
    height,
    name,
    color,
  } = props;

  const path = paths[name] || undefined;

  return (
    path ?
      <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 24 24"
        width={width}
        height={height}
        style={{ verticalAlign: 'middle' }}
      >
        <title>{name}</title>
        <path d={path} fill={color} />
        <polygon points={points} fillOpacity="0" />
      </svg>
      :
      null
  );
};

const defaultProps = {
  width: 24,
  height: 24,
  color: '#000000',
};

const numType = oneOfType([string, number]);

const propTypes = {
  /** Width & height can be number (converted to px) or string (e.g. 1rem) */
  width: numType,
  height: numType,
  /** The name of the icon, pulled from 'paths.js' */
  name: string.isRequired,
  color: string,
};

Icon.defaultProps = defaultProps;
Icon.propTypes = propTypes;
export default Icon;
