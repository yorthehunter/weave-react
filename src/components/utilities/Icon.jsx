import React from 'react';
import { oneOfType, number, string } from 'prop-types';
import iconPaths from './iconPaths';

export const Icon = (props) => {
  const {
    width,
    height,
    name,
    color,
  } = props;

  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      width={width}
      height={height}
      style={{ verticalAlign: 'middle' }}
    >
      <title>{name}</title>
      <path d={iconPaths[name]} fill={color} />
      <polygon points="0 0 24 0 24 24 0 24" fillOpacity="0" />
    </svg>
  );
};

const defaultProps = {
  width: 24,
  height: 24,
  color: '#000000',
};

const numType = oneOfType([string, number]);

const propTypes = {
  width: numType,
  height: numType,
  name: string.isRequired,
  color: string,
};

Icon.defaultProps = defaultProps;
Icon.propTypes = propTypes;
export default Icon;
