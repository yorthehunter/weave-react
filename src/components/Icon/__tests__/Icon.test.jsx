import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../';

describe('Icon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Icon name="close" />);
    const rendered = toJson(wrapper);
    expect(rendered).toMatchSnapshot();
  });

  it('renders null if name is not valid', () => {
    const wrapper = shallow(<Icon name="blobbityBoo" />);
    const rendered = toJson(wrapper);
    expect(rendered).toMatchSnapshot();
  });
});
