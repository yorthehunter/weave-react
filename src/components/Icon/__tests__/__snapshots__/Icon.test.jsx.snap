// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<svg
  height={24}
  style={
    Object {
      "verticalAlign": "middle",
    }
  }
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    close
  </title>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 Z"
    fill="#000000"
  />
  <polygon
    fillOpacity="0"
    points="0 0 24 0 24 24 0 24"
  />
</svg>
`;

exports[`Icon renders null if name is not valid 1`] = `""`;
